src/random_ray/random_ray.cpp

```diff
@@ -235,16 +235,21 @@ vector<double> rhalton(int dim, uint64_t* seed, int64_t skip = 0)
 // Static Variable Declarations
 double RandomRay::distance_inactive_;
 double RandomRay::distance_active_;
+double RandomRay::avg_miss_rate;
 unique_ptr<Source> RandomRay::ray_source_;
 RandomRaySourceShape RandomRay::source_shape_ {RandomRaySourceShape::FLAT};
 bool RandomRay::mesh_subdivision_enabled_ {false};
 RandomRaySampleMethod RandomRay::sample_method_ {RandomRaySampleMethod::PRNG};
+int64_t RandomRay::n_source_regions;
+int64_t RandomRay::n_external_source_regions;
+uint64_t RandomRay::total_geometric_intersections;
+int RandomRay::negroups_;
 
 RandomRay::RandomRay()
   : angular_flux_(data::mg.num_energy_groups_),
-    delta_psi_(data::mg.num_energy_groups_),
-    negroups_(data::mg.num_energy_groups_)
+    delta_psi_(data::mg.num_energy_groups_)
 {
+  negroups_ = data::mg.num_energy_groups_;
```

Comment:
As `negroups_` is now a shared global variable, all threads will be contending to write and rewrite to this variable in an uncontrolled manner in parallel. Perhaps this one can be left as a local variable? Then, we can just use the `data::mg.num_energy_groups_` variable directly when it is needed for writing out to the statepoint, if that's possible.  Longer term, it would be nice if all the random ray code just used one `negroups` variable somewhere, rather than each class having its own copy, but that is beyond the scope of this PR for sure.